### ISSUE TITLE
Accomodate for key being passed withing a v-bind object of props

### ISSUE THREAD

**Issue by RainbowGiantSquid** [vuejs/eslint-plugin-vue#2123]
**What rule do you want to change?**

https://eslint.vuejs.org/rules/valid-v-for.html

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
needs to accomodate for the key being passed in the v-bind object

**Please provide some example code that this change will affect:**
currently throwing a warning with missing key directive

```vue
<ComponentName v-for="item in array" v-bind="{key: item.id, anotherProp, oneMoreProp}" />

//another case:

<ComponentName v-for="(item, key) in array" v-bind="{key, anotherProp, oneMoreProp}" />

```
**What does the rule currently do for this code?**

throwing a 'require v-bind:key directive'

**What will the rule do after it's changed?**

Will not throw the warning when key is passed in an object with other props.

**Additional context**

**Comment by FloEdelmann** [comment]
I think it's fine if the value of `v-bind` is an object literal that contains a `key` item.
